packages/dashboard-frontend/src/services/workspace-client/devworkspace/devWorkspaceClient.ts

```diff
@@ -537,22 +551,33 @@ export class DevWorkspaceClient extends WorkspaceClient {
         env =>
           env.name !== this.dashboardUrlEnvName &&
           env.name !== this.pluginRegistryUrlEnvName &&
-          env.name !== this.pluginRegistryInternalUrlEnvName,
+          env.name !== this.pluginRegistryInternalUrlEnvName &&
+          env.name !== this.openVSXUrlEnvName,
       );
-      envs.push(
-        {
+      if (dashboardUrl) {
+        envs.push({
           name: this.dashboardUrlEnvName,
-          value: window.location.origin,
-        },
-        {
+          value: dashboardUrl,
+        });
+      }
+      if (pluginRegistryUrl) {
+        envs.push({
           name: this.pluginRegistryUrlEnvName,
           value: pluginRegistryUrl,
-        },
-        {
+        });
+      }
+      if (pluginRegistryInternalUrl) {
+        envs.push({
           name: this.pluginRegistryInternalUrlEnvName,
           value: pluginRegistryInternalUrl,
-        },
-      );
+        });
+      }
+      if (openVSXUrl) {
```

Comment:
Could you provide an empty environment variable

because on CheCode we need to know if it's empty or not

if it's missing, we don't know if it's empty or if we're running on previous che versions that are not providing the value ?

Reply:
so we should always have the env variable defined, just that it can contains a value or be empty

Reply:
I have fixed it

Reply:
![Знімок екрана 2022-08-22 о 17 00 52](https://user-images.githubusercontent.com/6310786/185939825-fa5d8e39-34e8-434d-99df-6055b45994bf.png)

Reply:
thanks @olexii4